website/addons/s3/static/s3UserConfig.js

```diff
@@ -0,0 +1,205 @@
+/**
+* Module that controls the S3 user settings. Includes Knockout view-model
+* for syncing data.
+*/
+
+var ko = require('knockout');
+require('knockout.punches');
+ko.punches.enableAll();
+var $ = require('jquery');
+var Raven = require('raven-js');
+var bootbox = require('bootbox');
+require('js/osfToggleHeight');
+
+var language = require('js/osfLanguage').Addons.s3;
+var osfHelpers = require('js/osfHelpers');
+var addonSettings = require('js/addonSettings');
+
+var ExternalAccount = addonSettings.ExternalAccount;
+
+var $modal = $('#s3InputCredentials');
+
+
+function ViewModel(url) {
+    var self = this;
+
+    self.properName = 'Amazon S3';
+    self.accessKey = ko.observable();
+    self.secretKey = ko.observable();
+    self.account_url = '/api/v1/settings/s3/accounts/';
+    // Whether the initial data has been loaded
+    self.loaded = ko.observable(false);
+    self.accounts = ko.observableArray();
+
+    // Flashed messages
+    self.message = ko.observable('');
+    self.messageClass = ko.observable('text-info');
+
+    /** Reset all fields from S3 credentials input modal */
+    self.clearModal = function() {
+        self.message('');
+        self.messageClass('text-info');
+        self.accessKey(null);
+        self.secretKey(null);
+    };
+
+    self.updateAccounts = function() {
+        var request = $.get(self.account_url);
+        request.done(function(data) {
+            self.accounts($.map(data.accounts, function(account) {
+                var externalAccount =  new ExternalAccount(account);
+                externalAccount.accessKey = account.oauth_key;
+                externalAccount.secretKey = account.oauth_secret;
+                return externalAccount;
+            }));
+            $('.addon-auth-table').osfToggleHeight({height: 140});
+        });
+        request.fail(function(xhr, status, error) {
+            Raven.captureMessage('Error while updating addon account', {
+                url: url,
+                status: status,
+                error: error
+            });
+        });
+        return request;
+    };
+
+    /** Send POST request to authorize S3 */
+    self.sendAuth = function() {
+        // Selection should not be empty
+        if( !self.accessKey() && !self.secretKey() ){
```

Comment:
It seems like the only way to do that and preserve all three messages would be to keep both the access and secret keys in the same observable, with the validator checking both to determine the correct message. This would mess up the data-binding in the template, however. I'm just going to leave it be for now.